internal/runtimehandlerhooks/utils_linux.go

```diff
@@ -204,20 +208,31 @@ func updateIrqBalanceConfigFile(irqBalanceConfigFile, newIRQBalanceSetting strin
 	return nil
 }
 
-func retrieveIrqBannedCPUMasks(irqBalanceConfigFile string) (string, error) {
+func retrieveIrqBannedCPUList(irqBalanceConfigFile string) (cpuset.CPUSet, error) {
 	input, err := os.ReadFile(irqBalanceConfigFile)
 	if err != nil {
-		return "", err
+		return cpuset.New(), err
 	}
 
 	lines := strings.Split(string(input), "\n")
 	for _, line := range lines {
-		if strings.HasPrefix(line, irqBalanceBannedCpus+"=") {
-			return strings.Trim(strings.Split(line, "=")[1], "\""), nil
+		// try to read from the legacy file first
+		// after the first iteration, it'll comment this line, so
+		// it will not enter this flow again (because the prefix won't match).
+		if strings.HasPrefix(line, irqBalanceBannedCpusLegacy+"=") {
+			list := strings.Trim(strings.Split(line, "=")[1], "\"")
+
+			return mapHexCharToCPUSet(list)
```

Comment:
What if both are present at the same time?

Reply:
It will preferred and go through the legacy flow first and only once. 

On the next iteration the legacy value will be commented out and the new value will be used.